### PR TITLE
Shayan/SmartChart not loaded in binary bot when language change to Espanyol

### PR DESCRIPTION
**Pull request by shayan-binary**
…rring to it

### CODE REVIEW

File: src/components/ChartSetting.jsx [hunk 1]
```diff
@@ -47,156 +47,165 @@ const ThemeToggle = ({
...
-                    <div className="ciq-list-item">
-                        <span className="ciq-icon-text">{t.translate('Auto Scale')}</span>
-                        <div className="ciq-action">
-                            <Switch
-                                value={isAutoScale}
-                                onChange={setAutoScale}
-                            />
+class ChartSetting extends Component {
+    componentDidMount() {
+        const { init, activeLanguages } = this.props;
```
Comment (morteza-binary) [inline src/components/ChartSetting.jsx]: @shayan-binary What happens by changing `activeLanguages` after mounting the component. Assume the page is loaded completely then the value of `activeLanguages` should be changed.

Comment (shayan-binary) [reply src/components/ChartSetting.jsx]: @morteza-binary so you mean. `activeLanguages` may be changed during the rendering?

Comment (shayan-binary) [reply src/components/ChartSetting.jsx]: of course, we can make it in a way to update per `activeLanguages` changed, I just thought it's going to initial in the beginning

Comment (morteza-binary) [reply src/components/ChartSetting.jsx]: @shayan-binary it'd be better to support it too.

File: src/store/ChartSettingStore.js [hunk 2]
```diff
@@ -104,37 +104,67 @@ export default class ChartSettingStore {
...
-               historical                  : this.historical,
-               language                    : this.language.key,
-               position                    : this.position,
-               isAutoScale                 : this.isAutoScale,
-               isHighestLowestMarkerEnabled: this.isHighestLowestMarkerEnabled,
-               theme                       : this.theme,
-           });
-       }
-   }
+    @action.bound updateProps(activeLanguages) {
```
Comment (morteza-binary) [inline src/store/ChartSettingStore.js]: @shayan-binary it should be like this
`@action.bound updateProps({ activeLanguages }) {`

**Pull request merged**